File: library/src/main/java/androidx/fragment/app/ReportFragmentManager.java
package androidx.fragment.app;

class ReportFragmentManager extends FragmentManager {

    boolean isShow = true;

    @Override
    void dispatchResume() {
        if (isShow) super.dispatchResume();
    }

    @Override
    void dispatchStart() {
        if (isShow) super.dispatchStart();
    }

    @Override
    void dispatchDestroyView() {
        if (isShow) super.dispatchDestroyView();
    }

    @Override
    void dispatchCreate() {
        if (isShow) super.dispatchCreate();
    }

    @Override
    void dispatchPause() {
        if (isShow) super.dispatchPause();
    }

    @Override
    void dispatchActivityCreated() {
        if (isShow) super.dispatchPause();
    }
}

File: library/src/main/kotlin/androidx/fragment/app/ReportFragmentManager.kt
package androidx.fragment.app

internal class ReportFragmentManager : FragmentManager() {

    private val reportFragment by lazy {
        parent as ReportFragment
    }

    private val isInForeground
        get() = reportFragment.isInForeground

    override fun dispatchResume() {
        if (isInForeground) super.dispatchResume()
    }

    override fun dispatchStart() {
        if (isInForeground) super.dispatchStart()
    }

    override fun dispatchCreate() {
        if (isInForeground) super.dispatchCreate()
    }

    override fun dispatchActivityCreated() {
        if (isInForeground) super.dispatchActivityCreated()
    }

    override fun dispatchPause() {
        if (isInForeground) super.dispatchPause()
    }

    override fun dispatchStop() {
        if (isInForeground) super.dispatchStop()
    }

    override fun dispatchDestroyView() {
        if (isInForeground) super.dispatchDestroyView()
    }

}